packages/wrangler/src/vectorize/insert.ts

```diff
@@ -30,13 +42,42 @@ export async function handler(
 	args: StrictYargsOptionsToInterface<typeof options>
 ) {
 	const config = readConfig(args.config, args);
+	const file = await open(args.file);
+
+	let index: Optional<VectorizeVectorMutation, "ids"> | undefined;
+	for await (const batch of getBatchFromFile(file, args.batchSize)) {
+		const formData = new FormData();
+		formData.append("vectors", new File([batch.join(`\n`)], "vectors.ndjson"));
+		logger.log(`✨ Uploading vector batch (${batch.length} vectors)`);
+		const idxPart = await insertIntoIndex(config, args.name, formData);
+		if (!index) index = idxPart;
+		else index.count += idxPart.count;
 
-	const vectors: VectorizeVector[] = [];
-	if (args.vectors) {
-		// Parse each vector into a Vector type
-		// Think about potential limits on args.vectors.length?
+		if (index.count > VECTORIZE_MAX_UPSERT_VECTOR_RECORDS) {
+			logger.warn(
+				`🚧 While Vectorize is in beta, we've limited uploads to 100k vectors`
+			);
+			logger.warn(
+				`🚧 You may run this again with another batch to upload further`
+			);
+			break;
+		}
 	}
 
-	const index = await insertIntoIndex(config, args.name, vectors);
+	// remove the ids - skip tracking these for bulk uploads since this could be in the 100s of thousands.
+	if (index) delete index.ids;
 	logger.log(JSON.stringify(index, null, 2));
 }
+
+// helper method that reads an ndjson file line by line in batches. not this doesn't
+// actually do any parsing - that will be handled on the backend
+async function* getBatchFromFile(file: FileHandle, batchSize = 3) {
+	let batch: string[] = [];
+	for await (const line of file.readLines()) {
```

Comment:
[`filehandle.readLines()`](https://nodejs.org/api/fs.html#filehandlereadlinesoptions) was added in Node 18—Wrangler supports Node 16, and so this won't be usable, I don't think. https://nodejs.org/api/readline.html is probably the best alternative?